pontoon/checks/utils/__init__.py

```diff
@@ -0,0 +1,48 @@
+from django.http import JsonResponse
+
+import compare
+import translate_toolkit
+
+
+def get_quality_checks(entity, locale, plural_form, string, ignore_warnings):
+    """
+    Main function that performs all quality checks from frameworks handled in Pontoon.
+
+    :arg pontoon.base.models.Entity entity: Source entity
+    :arg pontoon.base.models.Locale locale: Locale of a translation
+    :arg basestring string: a translation
+    :arg int plural_form: plural form of a translation
+    :arg bool ignore_warnings: omit warnings
+    :return: Return types:
+        * JsonResponse - If there are errors or
+        * None - If there's no errors and non-omitted warnings.
+    """
+    try:
+        cl_checks = compare.quality_check(entity, locale.code, string, plural_form)
+    except compare.UnsupportedResourceTypeError:
+        cl_checks = None
+
+    resource_ext = entity.resource.format
+
+    # Some of checks from compare-locales overlap checks from Translate Toolkit
+    tt_disabled_checks = set()
+
+    if cl_checks is not None:
+        if resource_ext == 'properties':
+            tt_disabled_checks = {
+                'escapes',
+                'nplurals',
+                'printf'
+            }
+
+    tt_checks = translate_toolkit.quality_check(entity.string, string, locale, tt_disabled_checks)
+
+    checks = dict(
+        tt_checks,
+        **(cl_checks or {})
+    )
+
+    if (not ignore_warnings and checks) or ('clErrors' in checks):
```

Comment:
The `ignore_warnings` flag should be removed in the future (but not as part of this patch). For now, its role should be changed to enable or disable Translate Toolkit checks only (because they yield a lot of false positives, whereas compare locales checks don't).

You should update the logic and docstring to reflect that. Because this code will always show translate toolkit checks if we have failed compare locales checks, regardless if they are enabled or disabled.

Reply:
So, warnings from `compare-locales` should also remove save button?

Reply:
No.

Reply:
Okay, then do we need a separate flag to ignore `warnings` from the compare-locales?

Reply:
No. :) `ignore_warnings` is really a `ignore_ttk_checks`.

We should prevent translations with `compare_locales` errors and allow strings with `compare_locales` warnings.